NetCoreMentoring/src/NetCoreMentoring.Core/Services/CategoryService.cs

```diff
@@ -24,5 +32,42 @@ public IEnumerable<Category> GetCategories()
         {
             return _mapper.Map<IEnumerable<Category>>(_context.Categories.AsEnumerable());
         }
+
+        public Category GetCategory(int categoryId)
+        {
+            var category = _context.Categories.Find(categoryId);
+
+            if (category == null)
+            {
+                //TODO: return error
+                throw new Exception();
+            }
+
+            return _mapper.Map<Category>(category);
+        }
+
+        public byte[] GetPicture(int categoryId)
+        {
+            return _context.Categories.Find(categoryId).Picture;
+        }
+
+        public void UpdatePicture(int categoryId, IFormFile newPicture)
```

Comment:
the database level should work with bytes. All asp.net related logic should be handles on higher levels

Reply:
but this is business logic layer

Reply:
ok, it's the business layer, but not UI level. In other words, it's better to limit passing dependencies across the class hierarchy to places where these dependencies are required. Asp.net related classes should be only on asp.net (which is mostly UI) level.